File: MyAPI/MyAPI/Data/PromotionInfo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace MyAPI.Data
{
    public class PromotionInfo
    {
        public int Id { get; set; }

        [ForeignKey(nameof(Product))]
        public int ProductId { get; set; }
        public Product Product { get; set; }

        [ForeignKey(nameof(Promotion))]
        public int PromotionId { get; set; }
        public Promotion Promotion { get; set; }
        public string PromotionPercent { get; set; }
        public string PromotionAmount { get; set; }
    }
}

File: MyFE/src/app/class/promotion-info.ts
import { Product } from "./product"
import { Promotion } from "./promotion"

export class PromotionInfo {

    id!:number
    product!:Product
    productId!:number
    promotionId!: number
    promotion!:Promotion
    promotionAmount!:string
    promotionPercent!: string
}
